prometheus/types.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2021 The go-commons Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prometheus
+
+import (
+	"net/http"
+
+	apiv1 "github.com/prometheus/client_golang/api/prometheus/v1"
+)
+
+// Prometheus describes the prometheus connection
+type Prometheus struct {
+	Api      apiv1.API
```

Comment:
With my previous comment this struct can embed the API object with: 

```suggestion
	apiv1.API
```

Thanks to this we could call the Prometheus functions w/o referencing the Api field

```golang
p.Client.QueryRange()
```

Reply:
As mentioned above, keeping the `API` private to this implementation and yes the downstreams can simply call `p.Client.QueryRange()` and `p.Client.Query()` to query for the results.